ddtrace/appsec/utils.py

```diff
@@ -13,13 +18,76 @@ def _appsec_rc_features_is_enabled():
 
 def _appsec_rc_capabilities():
     # type: () -> str
-    """return the bit of the composed capabilities in base64
+    r"""return the bit representation of the composed capabilities in base64
     bit 0: Reserved
-    bit 1: ASM Activation
+    bit 1: ASM 1-click Activation
     bit 2: ASM Ip blocking
 
-    TODO: refactor to compose the string and encode it
+    Int Number  -> binary number    -> bytes representation -> base64 representation
+    ASM Activation:
+    2           -> 10               -> b'\x02'              -> "Ag=="
+    ASM Ip blocking:
+    4           -> 100              -> b'\x04'              -> "BA=="
+    ASM Activation and ASM Ip blocking:
+    6           -> 110              -> b'\x06'              -> "Bg=="
+    ...
+    256         -> 100000000        -> b'\x01\x00'          -> b'AQA='
     """
+    value = 0b0
+
     if _appsec_rc_features_is_enabled():
-        return "Ag=="
-    return ""
+        value |= 1 << 1
+        value |= 1 << 2
+
+    result = ""
+    if sys.version_info.major < 3:
+        bytes_res = to_bytes_py2(value, (value.bit_length() + 7) // 8, "big")
+        result = base64.b64encode(bytes_res)  # type: ignore[assignment, arg-type]
+    else:
+        result = str(base64.b64encode(value.to_bytes((value.bit_length() + 7) // 8, "big")), encoding="utf-8")
+
+    return result
+
+
+_HTML_USERDEFINED_BLOCKED_TEMPLATE_CACHE = None  # type: Optional[str]
+_JSON_USERDEFINED_BLOCKED_TEMPLATE_CACHE = None  # type: Optional[str]
+
+
+def _get_blocked_template(accept_header_value):
+    # type: (str) -> str
+
+    global _HTML_USERDEFINED_BLOCKED_TEMPLATE_CACHE
+    global _JSON_USERDEFINED_BLOCKED_TEMPLATE_CACHE
+
+    need_html_template = False
+
+    if accept_header_value and "text/html" in accept_header_value.lower():
+        need_html_template = True
+
+    if need_html_template:
+        template_path = os.getenv("DD_APPSEC_HTTP_BLOCKED_TEMPLATE_HTML")
+    else:
+        template_path = os.getenv("DD_APPSEC_HTTP_BLOCKED_TEMPLATE_JSON")
+
+    if template_path and os.path.exists(template_path) and os.path.isfile(template_path):
+        if need_html_template and _HTML_USERDEFINED_BLOCKED_TEMPLATE_CACHE:
+            return _HTML_USERDEFINED_BLOCKED_TEMPLATE_CACHE
+
+        if not need_html_template and _JSON_USERDEFINED_BLOCKED_TEMPLATE_CACHE:
+            return _JSON_USERDEFINED_BLOCKED_TEMPLATE_CACHE
```

Comment:
could this check be moved up, like to L66

if we have the cached values, we don't need to get the env vars, or check if the files exist, etc.

Reply:
Not in 66 since we need to check the context type to know if we must return json or html, but could be moved after that (line 69).